app/utils/response_router.js

```diff
@@ -205,24 +226,34 @@ module.exports = {
   AWAITING_CAPTURE_REQUEST: {
     view: 'errors/charge_confirm_state_completed',
     locals: {status: 'successful'},
-    analyticsPage: '/success_return'
-  },
+    analyticsPage: '/success_return',
+    terminal: true
+  }
+}
 
-  display: function (res, viewName, options) {
-    let action = lodash.result(this, viewName)
-    options = options || {}
-    options.viewName = viewName
-    if (!action) {
-      logger.error('VIEW ' + viewName + ' NOT FOUND')
-      options = {viewName: 'error'}
-      action = this.ERROR
-    }
-    options = (action.locals) ? lodash.merge({}, action.locals, options) : options
+exports.response = (req, res, actionName, options) => {
+  options = options || {}
+  options.viewName = actionName
+  let action = lodash.result(actions, actionName)
+  if (!action) {
+    logger.error('Response action ' + actionName + ' NOT FOUND')
+    options = {viewName: 'error'}
+    action = actions.ERROR
+  }
 
-    if (lodash.get(options, 'analytics.path')) {
-      options.analytics.path = options.analytics.path + lodash.get(action, 'analyticsPage', '')
-    }
-    res.status(action.code || 200)
-    res.render(action.view, options)
+  const directRedirectEnabled = lodash.get(res, 'locals.service.redirectToServiceImmediatelyOnTerminalState', false)
+  if (action.terminal && directRedirectEnabled && req.chargeData) {
```

Comment:
I would be tempted to roll all of this into a function
```
shouldRedirect = (req, res, action) => {
    const directRedirectEnabled = lodash.get(res, 'locals.service.redirectToServiceImmediatelyOnTerminalState', false)
    return (action.terminal && directRedirectEnabled && req.chargeData)
}
```

But no biggie - maybe do in a separate PR if you agree